## Hello folks,This is [Shiva](https://www.linkedin.com/in/shivavashishtha/)!

Data Analytics professional, Extensive experience with Excel, SQL, Python, Tableau, and Power BI, analyzing large datasets, and generating actionable insights. Exceptional problem solver and quick learner with extensive experience in a variety of data, domains, and D&A technologies. 

Welcome to this Notebook,This notebook is a part of our Data Orinted Program. Scrolldown the notebook to learn more about the concept. If you face any issues feel free to reach out to me over [Linkedin](https://www.linkedin.com/in/shivavashishtha/)


<img align="center" alt="GIF" src="https://frogdesign.nyc3.cdn.digitaloceanspaces.com/wp-content/uploads/2020/08/04192430/AI_designing-with-data.gif">
<br>

<div class="alert alert-info" style="background-color:#FFFFFF; color:#808080; padding:0px 10px; border-radius:5px;" align="center"><h1 style='margin:12px 5px'>Measures of Central Tendency</h1>
</div>

Measures of central tendency are used in EDA (Exploratory Data Analysis) to describe the central or typical value of a data set. These measures provide a way to summarize the data and make it easier to understand.
Here are three common measures of central tendency:
Mean: The mean is the arithmetic average of a data set. It is calculated by adding up all the values in the data set and dividing by the total number of values. The mean is sensitive to outliers, which means that it can be influenced by extreme values in the data set.

1. Median: 
The median is the middle value in a data set. To find the median, the data set is first arranged in order from smallest to largest. If there is an odd number of values in the data set, the median is the middle value. If there is an even number of values, the median is the average of the two middle values. The median is less sensitive to outliers than the mean, which makes it a better measure of the central tendency for skewed data sets.

2. Mode: 
The mode is the most frequent value in a data set. It is the value that occurs with the highest frequency. A data set can have multiple modes if there are multiple values with the same frequency. The mode is useful for identifying the most common value or values in a data set.

In summary, measures of central tendency provide information about the typical or central value of a data set. The mean, median, and mode are commonly used measures of central tendency, each with its own advantages and disadvantages.

**Example:**
Let's say we want to understand the typical income of employees in a company. We can use measures of central tendency to summarize the income data.

For example, we can calculate the mean income by adding up all the individual salaries and dividing by the total number of employees. This will give us the average income of employees in the company.

Next, we can calculate the median income by ordering all the individual salaries from lowest to highest and finding the middle value. This will give us an idea of the income level that separates the top 50% of earners from the bottom 50% of earners.

Finally, we can calculate the mode income by identifying the most frequent salary in the data set. This will tell us the salary level that occurs most often among the employees in the company. By using these measures of central tendency, we can gain a better understanding of the income distribution of employees in the company and identify any outliers or anomalies in the data set.



## Importing the Libraries

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/iris/Iris.csv
/kaggle/input/iris/database.sqlite


## Reading the Data

In [2]:
df=pd.read_csv("/kaggle/input/iris/Iris.csv")

## Checking the Data

In [3]:
df

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
0      1            5.1           3.5            1.4           0.2   
1      2            4.9           3.0            1.4           0.2   
2      3            4.7           3.2            1.3           0.2   
3      4            4.6           3.1            1.5           0.2   
4      5            5.0           3.6            1.4           0.2   
..   ...            ...           ...            ...           ...   
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
0       Iris-setosa  
1       Iris-setosa  
2       Iris-setosa  
3       Iris-setosa  
4       Iris-setosa  
..              ...  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica  

[150 rows x 6 columns]

## Checking the Shape of the Data

In [4]:
df.shape

(150, 6)

The Dataset have 150 rows and 6 columns 

## Checking the info of the Data

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


This info analysis generally helps us in undestanding 3 major point. The index range, DataTypes of Features and Memory Usage of Data

## Checking the Describe feature

In [6]:
df.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

## Calculating the Mean

In [7]:
df.columns

Index(['Id', 'SepalLengthCm', 'SepalWidthCm', 'PetalLengthCm', 'PetalWidthCm',
       'Species'],
      dtype='object')

In [8]:
df.SepalLengthCm.mean()

5.843333333333334

In [9]:
df["SepalLengthCm"].mean()

5.843333333333334

In [10]:
df.SepalLengthCm.median()

5.8

In [11]:
df.SepalLengthCm.mode()

0    5.0
dtype: float64

## Calculating Mean from Scratch

In [12]:
#calculating sum of all the values present in the column
sum_values = df['SepalLengthCm'].sum()

#checking the number of rows or values
n_values = df['SepalLengthCm'].shape[0]

#simply diving the sum by number of values 
mean_value = sum_values/n_values

#printing the final result
print("Mean : ", mean_value)

Mean :  5.843333333333334


## Calculating Median from Scratch

So, As we know that the median is nothing but the middle value of Data when the Dataset is arranged in ascending or descending order

In [13]:
# calculate median from scratch
ordered_values = df['SepalLengthCm'].sort_values().reset_index(drop = True)
n_values = df['SepalLengthCm'].shape[0]

if n_values % 2 == 0:
    median1 = ordered_values[n_values//2]
    median2 = ordered_values[n_values//2 - 1]
    median_value = (median1 + median2)/2
else:
    median_value = ordered_values[n_values//2]

print("Mean : ", median_value)

Mean :  5.8


## Calculating Mode from Scratch

In [14]:
mode_value = df['SepalLengthCm'].value_counts().to_list()[0]
print("Mode : ", mode_value)

Mode :  10
